node/full.go

```diff
@@ -313,6 +319,70 @@ func (n *FullNode) AppClient() abciclient.Client {
 	return n.appClient
 }
 
+type ResultDirectTx struct {
+	Included bool   `json:"included"`
+	Height   uint64 `json:"height"`
+}
+
+func (n *FullNode) ReceiveDirectTx() func([]byte) ResultDirectTx {
+	// returns "true" if a block is built, "false" if it times out.
+	return func(tx []byte) ResultDirectTx {
+		n.Logger.Info("Receive Direct Tx")
+		n.incomingTxCh <- tx
```

Comment:
i don't think you need to do back-and-forth from node to manager just for inserting the received txn to mempool. why not directly call `validateTx`, instead of sending it to `incomingTxCh` and then expect the block manager to call `validateTx`?
also, directly receive the `error` from `validateTx` and indicate the status in the `ResultDirectTx`, instead of processing various errors and returning `true/false`

Reply:
great catch